.github/workflows/test-s3gw.yml

```diff
@@ -35,145 +31,54 @@ jobs:
           path: s3gw
           submodules: false
 
-      - name: Checkout s3tests
-        uses: actions/checkout@v3
-        with:
-          repository: ceph/s3-tests
-          path: s3tests
-
-      - name: Checkout s3gw-status
-        uses: actions/checkout@v3
-        with:
-          repository: ${{ github.repository_owner }}/s3gw-status
-          path: s3gw-status
-
-      - name: Configure Git
-        run: |
-          git config --global user.name "GitHub Actions Bot"
-          git config --global user.email "actions-bot@users.noreply.github.com"
-
-      - name: Setup Python
-        uses: actions/setup-python@v4
-        with:
-          python-version: 3.8
-
-      - name: Install Dependencies
-        run: |
-          YQ_GH_URL=https://github.com/mikefarah/yq/releases/download
-          YQ_VERSION=v4.31.1
-          YQ_CHECKSUM=1aef844cbecbbf81036449ea5e7dfcdf19d2d374fab6303fdb8f849d04275d76
-
-          sudo apt-get update
-          sudo apt-get install -y \
-            bc \
-            wget \
-            s3cmd
-
-          # Unfortunately, since yq is only available through snap on Ubuntu and
-          # that doesn't work in docker containers (at least not out of the
-          # box), this abomination is the way to go to install yq.
-          echo "${YQ_CHECKSUM}  yq" >> checksum
-          wget -O yq "${YQ_GH_URL}/${YQ_VERSION}/yq_linux_amd64"
-          sha256sum -c checksum \
-            && sudo mv yq /usr/bin/yq \
-            && sudo chmod +x /usr/bin/yq
-
-          python3 -m pip install -r s3tests/requirements.txt
-          python3 -m pip install -r ceph/qa/rgw/store/sfs/tests/requirements.txt
-
-      - name: Build Unittests
-        run: |
-          docker build \
-            --build-arg CMAKE_BUILD_TYPE=Debug \
-            --build-arg NPROC=16 \
-            --file s3gw/Dockerfile \
-            --tag s3gw-unittests \
-            --target s3gw-unittests \
-            .
-
-          docker images
-
-      - name: Run Unittests
-        run: |
```

Comment:
The two-stage build approach was deliberately chosen to also run unit tests. I'd not like to lose those. We're relying on Docker caching here to avoid building twice.